File: KickStart/starter_code/starter_code.py
def find_max_num_balanced_substrings(l, r):
  max_num_substrings = 0
  # TODO: Add logic to find the maximum possible number of balanced non-empty
  # substrings
  return max_num_substrings


def main():
  # Get number of test cases
  test_cases = int(raw_input())
  for test_case in range(1, test_cases + 1):
    # Get total of left and right parentheses
    l, r = map(int, raw_input().split())
    ans = find_max_num_balanced_substrings(l, r)
    print("Case #%d: %d" % (test_case, ans))


if __name__ == "__main__":
  main()

File: KickStart/starter_code/starter_code.java
import java.util.*;

public class Solution {

  private static long findMaxNumOfBalancedSubstrings(int l, int r) {
    long maxNumOfSubstrings = 0;
    // TODO: Add logic to find the maximum possible number of balanced non-empty substrings
    return maxNumOfSubstrings;
  }

  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    // Input number of test cases
    int testCaseCount = in.nextInt();
    for (int tc = 1; tc <= testCaseCount; ++tc) {
      // Input total number of left and right parentheses
      int l = in.nextInt();
      int r = in.nextInt();

      System.out.println("Case #" + tc + ": " + findMaxNumOfBalancedSubstrings(l, r));
    }
  }
}
